File: Frontend/src/types/academic-term.ts
export interface AcademicTermResponse {
  id: string;
  code: string;
  startsOn: string;
  endsOn: string;
}

export interface CreateAcademicTermRequest {
  code: string;
  startsOn: string;
  endsOn: string;
}

export interface UpdateAcademicTermRequest {
  code: string;
  startsOn: string;
  endsOn: string;
}

File: Backend/OnnorokomBackend/Models/Entities/AcademicTerm.cs
namespace OnnoRokomBackend.Models.Entities;

public class AcademicTerm
{
    public Guid Id { get; set; }
    public string Code { get; set; } = string.Empty;
    public DateOnly StartsOn { get; set; }
    public DateOnly EndsOn { get; set; }

    public ICollection<AcademicBatch> Batches { get; set; } = [];
}
